backend/data/program-credentials/merge-credentials.py

```diff
@@ -32,58 +32,47 @@ def merge_lookup_label(result: pd.DataFrame, lookup_df: pd.DataFrame, column: st
     result.insert(result.columns.get_loc(column)+1, f"{column}{lookup_label_column}", label_col)
     return result
 
-
-def addCredentials(maindf):
-    df = maindf
-    # new column credential type
-    df['CREDENTIALTYPE'] = ''
-
+def clean_official_name(df: pd.DataFrame):
     # replace special characters " and *
-    df['OFFICIALNAME'] = df['OFFICIALNAME'].str.replace('[*,"]', '')
+    df['OFFICIALNAME'] = df['OFFICIALNAME'].str.replace('[*,"]', '', regex=True)
     df['OFFICIALNAME'] = df['OFFICIALNAME'].str.replace('AAS', 'A.A.S.')
+    return df
+
 
+def label_credential_type(row: pd.Series):
     # Certification
-    df.loc[df['LEADTOINDUSTRYCREDENTIAL'] == "1", 'CREDENTIALTYPE'] = 'Certification'
+    official_name : str = row['OFFICIALNAME']
 
-    # GED
-    df.loc[df['OFFICIALNAME'].str.contains('GED'), 'CREDENTIALTYPE'] = 'General Education Diploma (GED)'
-    df.loc[df['OFFICIALNAME'].str.contains(
-        'General Education Diploma'), 'CREDENTIALTYPE'] = 'General Education Diploma (GED)'
-    df.loc[df['OFFICIALNAME'].str.contains(
-        'High School Equivalence'), 'CREDENTIALTYPE'] = 'General Education Diploma (GED)'
-    df.loc[df['OFFICIALNAME'].str.contains(
-        'High School Equivalency'), 'CREDENTIALTYPE'] = 'General Education Diploma (GED)'
+    if row['LEADTOLICENSE'] == "1" and row['LEADTOINDUSTRYCREDENTIAL'] == "0":
+        return 'License'
 
-    # Pre-Apprenticeship
-    df.loc[df['OFFICIALNAME'].str.contains('Pre-Apprentice'), 'CREDENTIALTYPE'] = 'Pre-Apprenticeship'
+    if 'M.' in official_name and 'Master' in row['DEGREEAWARDEDNAME']:
+        return 'Masters Degree'
 
-    # Apprenticeship
-    df.loc[(df['OFFICIALNAME'].str.contains('Apprentice')) & (
-                df['CREDENTIALTYPE'] != 'Pre-Apprenticeship'), 'CREDENTIALTYPE'] = 'Apprenticeship'
+    if 'A.S.' in official_name:
+        return 'Associate Degree'
 
-    # Secondary School Diploma
-    df.loc[df['OFFICIALNAME'].str.contains(
-        'High School Diploma - Adults'), 'CREDENTIALTYPE'] = 'Secondary School Diploma'
+    ESL_TOKENS = { 'ESL', 'English as a Second Language'}
```

Comment:
Should this be `ESL_NAME_TOKENS` to be consistent with next token?

Reply:
👍 Fixed up ESL_TOKENS to ESL_NAME_TOKENS.